### PR TITLE
[JN-360] Model DSM kit requests and skeleton service

### PR DESCRIPTION
**Pull request by breilly2**
Basic model of kit request and DAO, service, etc. Gives us a starting point to build on as we refine the details.

To test:
1. Start admin API
2. Open http://localhost:8080/swagger-ui.html#/enrollee/requestKit, authenticate, and fill in parameters (suggested values attached)
3. Check the database and verify that there is a new row in the `kit_request` table

Screen Shot 2023-05-18 at 2 34 52 PM

`kitType` is an uncontrolled string at the moment, so you can use anything you like.

### CODE REVIEW

File: core/src/main/java/bio/terra/pearl/core/service/participant/EnrolleeService.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ public Enrollee create(Enrollee enrollee) {
             enrollee.setShortcode(generateShortcode());
         }
         Enrollee savedEnrollee = dao.create(enrollee);
-        logger.info("Enrollee created.  id: {}, shortcode: {}, participantUserId", savedEnrollee.getId(),
+        logger.info("Enrollee created.  id: {}, shortcode: {}, participantUserId: {}", savedEnrollee.getId(),
```
Comment (breilly2) [inline core/src/main/java/bio/terra/pearl/core/service/participant/EnrolleeService.java:152]: Drive-by fixing a bug that I discovered while reading code.

File: core/src/test/java/bio/terra/pearl/core/service/participant/EnrolleeServiceTests.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public void testEnrolleeCreate() {
         Enrollee savedEnrollee = enrolleeService.create(enrollee);
         DaoTestUtils.assertGeneratedProperties(savedEnrollee);
         Assertions.assertNotNull(savedEnrollee.getShortcode());
-        Assertions.assertEquals(enrollee.getParticipantUserId(), enrollee.getParticipantUserId());
+        Assertions.assertEquals(enrollee.getParticipantUserId(), savedEnrollee.getParticipantUserId());
```
Comment (breilly2) [inline core/src/test/java/bio/terra/pearl/core/service/participant/EnrolleeServiceTests.java:37]: Another drive-by bug fix.

File: api-admin/src/main/resources/api/openapi.yml [hunk 3]
```diff
@@ -298,6 +298,25 @@ paths:
...
+    post:
+      summary: Request a sample kit for an enrollee
+      tags: [ enrollee ]
+      operationId: requestKit
+      parameters:
+        - { name: portalShortcode, in: path, required: true, schema: { type: string } }
+        - { name: studyShortcode, in: path, required: true, schema: { type: string } }
+        - { name: envName, in: path, required: true, schema: { type: string } }
+        - { name: enrolleeShortcode, in: path, required: true, schema: { type: string } }
+      requestBody:
```
Comment (devonbush) [inline api-admin/src/main/resources/api/openapi.yml]: Since kitType is required and a string, and fundamental to the object, I'd lean towards making kitType a URL param, probably on the path.  The big advantage there is it then gives us easier logging/analytics since all API requests are logged

File: core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                mailingAddress.getStreet1(),
+                mailingAddress.getStreet2(),
+                mailingAddress.getCity(),
+                mailingAddress.getState(),
+                mailingAddress.getPostalCode(),
+                mailingAddress.getCountry(),
+                profile.getPhoneNumber()
+        );
+    }
+
```
Comment (devonbush) [inline core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java]: nit: I wouldn't pull this out into its own method unless you expect to be doing something besides objectMapping soon.

File: core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                .status(KitRequestStatus.CREATED)
+                .build();
+        KitRequest savedKitRequest = dao.create(kitRequest);
+        logger.info("SampleKit created. id: {}, enrollee: {}", savedKitRequest.getId(), savedKitRequest.getEnrolleeId());
+        return savedKitRequest;
+    }
+
+    private PepperKitAddress makePepperKitAddress(Profile profile) {
+        MailingAddress mailingAddress = profile.getMailingAddress();
+        return new PepperKitAddress(
```
Comment (devonbush) [inline core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java]: Long order-sensitive string constructions like this make me nervous.  This makes me lean towards making PepperKitAddress a class so you can use a lombok builder.  I've had to abandon using Records elsewhere for this same reason

File: core/src/main/resources/db/changelog/changesets/2023_05_15_kit_request.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+databaseChangeLog:
+  - changeSet:
+      id: kit_request
+      author: breilly
+      changes:
+        - createTable:
+            tableName: kit_request
+            columns:
```
Comment (devonbush) [inline core/src/main/resources/db/changelog/changesets/2023_05_15_kit_request.yaml:8]: BLOCKER: we need a "creating_admin_user_id" field here for auditability.  For now, it should be non-nullable.  Later, when we add rule-based auto-kit sending, we can modify that constraint to say "must have either an admin id, or a rule_source specified"

File: core/src/testFixtures/java/bio/terra/pearl/core/factory/participant/EnrolleeFactory.java [hunk 7]
```diff
@@ -46,10 +46,11 @@ public Enrollee buildPersisted(String testName) {
         return enrolleeService.create(builderWithDependencies(testName).build());
     }
 
-    public Enrollee buildPersisted(String testName, UUID studyEnvironmentId, UUID participantUserId) {
+    public Enrollee buildPersisted(String testName, UUID studyEnvironmentId, UUID participantUserId, UUID profileId) {
```
Comment (devonbush) [inline core/src/testFixtures/java/bio/terra/pearl/core/factory/participant/EnrolleeFactory.java:49]: maybe instead of adding a paramter to this function, create a new one, and have the existing one delegate?  So we don't have to specify "null" in a bunch of places where we don't care about the profile?

**Review by devonbush — CHANGES_REQUESTED** [review]
Good start.  A bunch of suggestions that aren't necessarily blocking, and one blocker around recording the admin user who initiated the kitrequest

File: core/src/testFixtures/java/bio/terra/pearl/core/factory/participant/EnrolleeFactory.java (hunk 7 above)
Comment (devonbush) [reply core/src/testFixtures/java/bio/terra/pearl/core/factory/participant/EnrolleeFactory.java:49]: nvm, it looks like this wasn't used elsewhere

File: core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java (hunk 5 above)
Comment (breilly2) [reply core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java]: 👍 Good call.

**Review by devonbush — APPROVED** [review]
nice

**Review by MatthewBemis — APPROVED** [review]
